packages/remote-config/src/abt/experiment.ts

```diff
@@ -76,10 +78,14 @@ export class Experiment {
     customProperty: Record<string, string | null>
   ): Promise<void> {
     try {
-      const analytics = await this.analyticsProvider.get();
-      analytics.setUserProperties({ properties: customProperty });
+      const analytics = this.analyticsProvider.getImmediate({ optional: true });
+      if (analytics) {
+        analytics.setUserProperties({ properties: customProperty });
+      } else {
+        this.logger.warn(`Analytics is not imported correctly`);
+      }
     } catch (error) {
-      console.error(`Failed to add experiment to analytics :`, error);
+      this.logger.error(`Failed to add experiment to analytics : ${error}`);
```

Comment:
Just want to confirm it's a deliberate choice to log an error but not to throw?

Reply:
added a new error type

Reply:
Should probably remove the logged error if throwing an error?